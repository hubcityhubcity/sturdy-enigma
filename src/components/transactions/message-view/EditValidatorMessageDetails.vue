<template>
  <div>
    <div class="tx__content__caption">
      <p>Edit validator</p>
    </div>
    <div class="tx__content__information">
      Monikor&nbsp;
      <router-link
        :to="`staking/validators/${transaction.value.validator_address}`"
      >
        {{
          transaction.value.validator_address | resolveValidatorName(validators)
        }}
      </router-link>
    </div>
  </div>
</template>

<script>
import { atoms, viewDenom, prettyLong } from "scripts/num.js"
import { resolveValidatorName } from "src/filters"

export default {
  name: `edit-validator-message-details`,
  filters: {
    atoms,
    viewDenom,
    prettyLong,
    resolveValidatorName
  },
  props: {
    transaction: {
      type: Object,
      required: true
    },
    validators: {
      type: Object,
      required: true
    }
  }
}
</script>
