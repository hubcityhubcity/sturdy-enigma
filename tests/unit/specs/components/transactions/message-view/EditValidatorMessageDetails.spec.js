import { shallowMount } from "@vue/test-utils"
import EditValidatorMessageDetails from "src/components/transactions/message-view/EditValidatorMessageDetails"

describe(`EditValidatorMessageDetails`, () => {
  let wrapper
  // Examples @ https://stargate.cosmos.network/txs?action=edit_validator
  const tx = {
    height: 2000000,
    value: {
      address: `cosmos1`
    },
    fee: {
      amount: [
        {
          amount: "37",
          denom: "uatom"
        }
      ]
    },
    hash: `ABCD1234`,
    group: `staking`,
    key: "keyhash",
    memo: "(Sent via Lunie)",
    timestamp: 123456789,
    liquidDate: null,
    type: `cosmos-sdk/MsgEditValidator`
  }

  it(`renders a edit validator transaction message`, () => {
    wrapper = shallowMount(EditValidatorMessageDetails, {
      propsData: {
        transaction: tx,
        validators: {}
      },
      stubs: [`router-link`]
    })
    expect(wrapper.element).toMatchSnapshot()
  })
})
