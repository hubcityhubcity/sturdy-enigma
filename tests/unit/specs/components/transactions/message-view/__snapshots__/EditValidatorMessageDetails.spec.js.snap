// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditValidatorMessageDetails renders a edit validator transaction message 1`] = `
<div
  class="tx__content"
>
  <transactionicon-stub
    transactiongroup="staking"
    transactiontype="Edit"
  />
   
  <div
    class="tx__content__left"
  >
    <h3>
      Edit validator
    </h3>
     
    <span>
      moniker 
    </span>
     
    <router-link-stub
      to="/staking/validators/cosmos1"
    >
      <!---->
      
      cosmos…mos1
    
    </router-link-stub>
  </div>
   
  <div
    class="tx__content__right"
  >
    <p
      class="amount"
    >
      
      0.000037 
      ATOM
    
    </p>
  </div>
</div>
`;
